thorlcr/activities/loop/thloop.cpp

```diff
@@ -320,9 +321,17 @@ class CGraphLoopActivityMaster : public CLoopActivityMasterBase
         unsigned loopCounter = 1;
         for (;;)
         {
+            Owned<IThorGraphResults> results = queryGraph().createThorGraphResults(1);
+            unsigned condLoopCounter = (helper->getFlags() & IHThorGraphLoopArg::GLFcounter) ? loopCounter : 0;
+            IThorBoundLoopGraph *boundGraph = queryContainer().queryLoopGraph();
+            if (condLoopCounter)
+                boundGraph->prepareCounterResult(*this, results, condLoopCounter, 0);
```

Comment:
Should the last parameter be a 2 to match process()?

Reply:
This is as it was before but it was a bit confusing.
It is 2 in the LOOP case, and 0 in the GRAPH case.

And I'm hoping that regression suite tests are validating it is correct.
Having had a quick look at roxie code, I think 0 for GRAPH and 2 for LOOP is correct or at least same in Roxie. Can you verify?

The code above is for GRAPH and before this change it used to call one of the 2nd form of IThorBoundLoopGraph::execute which used to set the counter result as result 0.